cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1169,9 +1223,11 @@ build_chunk_dictionaries(hostdevice_2dvector<EncColumnChunk>& chunks,
   std::vector<rmm::device_uvector<slot_type>> hash_maps_storage;
   hash_maps_storage.reserve(h_chunks.size());
   for (auto& chunk : h_chunks) {
-    if (col_desc[chunk.col_desc_id].physical_type == Type::BOOLEAN ||
-        (col_desc[chunk.col_desc_id].output_as_byte_array &&
-         col_desc[chunk.col_desc_id].physical_type == Type::BYTE_ARRAY)) {
+    auto const& chunk_col_desc = col_desc[chunk.col_desc_id];
+    if (chunk_col_desc.physical_type == Type::BOOLEAN ||
+        (chunk_col_desc.output_as_byte_array && chunk_col_desc.physical_type == Type::BYTE_ARRAY) ||
+        (chunk_col_desc.requested_encoding != Encoding::UNDEFINED &&
```

Comment:
This condition has become hard(er) to parse. Maybe having a local `bool is_non_requested_dictionary` would help.